sonar-ws/src/main/java/org/sonarqube/ws/client/WsClientFactory.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarqube.ws.client;
+
+import org.sonar.api.server.ws.LocalConnector;
+
+/**
+ * @since 5.5
+ */
+public class WsClientFactory {
```

Comment:
I don't agree on the need. sonar-ws is a library and by design it can't be provided at runtime as sonar-plugin-api. The constraints are not the same at all. In this context having `WsClientFactory.getDefault().newClient(HttpConnector.builder()....build())` is too complex. Let's keep things simple and do not overthink about future needs.